docs/docs/learn/shards/coding-with-shards.md

```diff
@@ -0,0 +1,220 @@
+# Coding with Shards
+
+In this chapter, we will be learning how to code with Shards so that you can write your very own program!
+
+## Data Types
+
+A shard can take in data, process it, and output the results.
+
+There are many different types of data in the Shards language, and each shard will have specific data types that it can work with. 
+
+For example, the [`Math.Add`](../../../reference/shards/Math/Add/) shard can only work with numeric data types, while the [`Log`](../../../reference/shards/General/Log/) shard that prints information to the console can work with `Any` data type.
+
+Here are some of the data types found in Shards:
+
+| Data Type   | Description                                            | Example         |
+| :---------- | :----------------------------------------------------- | :-------------- |
+| `Any`       | Any data type.                                         |                 |
+| `Bool`      | Evaluates to either `true` or `false`.                 | `true`, `false` |
+| `Float`     | A numerical value with a decimal point.                | `2.53`, `9.124` |
+| `Int`       | A numerical value with no decimals. Read as "integer". | `2`, `9`        |
+| `None`      | No data type.                                          |                 |
+| `Sequence`  | A collection of values.                                | `[3, 25, 9]`    |
+| `String`    | Characters enclosed by double quotes.                  | "A string!"     |
+| `Wire`      | A sequence of shards.                                  |                 |
+
+For the full list of data types and more in-depth reading, check out the `Types` documentation page [here](../../../reference/shards/types/).
+
+## Variables
+
+To better work with data across your code, we can assign them to data containers known as *variables*. Imagine a scenario where you have a float `3.141592653589793238` that you need to reuse in code multiple times. Instead of typing the entire float out each time, you could assign it to a variable called `.pi-value` and simply use that variable whenever its needed.
+
+=== "Without Variables"
+
+    ```{.clojure .annotate linenums="1"}
+    3.141592653589793238 (Math.Add 3.141592653589793238) (Math.Multiply 3.141592653589793238) (Math.Subtract 3.141592653589793238)
+    ```
+=== "With Variables"
+
+    ```{.clojure .annotate linenums="1"}
+    3.141592653589793238 = .pi-value ;; (1)
+    .pi-value (Math.Add .pi-value) (Math.Multiply .pi-value) (Math.Subtract .pi-value)
+    ```
+
+    1. 3.141592653589793238 is assigned to the variable `.pi-value`. We'll learn more about assigning variables in a bit!
+  
+Variable names always start with a `.` period.
+
+Some example of variable names:
+
+- `.x`
+
+- `.number-of-apples`
+
+- `.is-verified` 
+
+How you assign data to variables depends on the variable type. The main difference between variables are as follows:
```

Comment:
Either `differences` and `are`  or `difference` and `is`. I believe plural here is warranted.

```suggestion
How you assign data to variables depends on the variable type. The main differences between variables are as follows:
```